app/models/post.rb

```diff
@@ -0,0 +1,19 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :likes, class_name: 'Like', dependent: :destroy
+  has_many :comments, dependent: :destroy
+
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
+  after_save :update_post_counter
```

Comment:
Great job @AlyAtalla 🤝

- [ ] [Optional] Since you have used `after_save` for the `update_post_counter`, you may also make this method private. Kindly repeat this for related methods in the `like` and `comment` models.